x-pack/test/alerting_api_integration/observability/threshold_rule_data_view.ts

```diff
@@ -0,0 +1,132 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { Aggregators, Comparator } from '@kbn/observability-plugin/common/threshold_rule/types';
+import { OBSERVABILITY_THRESHOLD_RULE_TYPE_ID } from '@kbn/observability-plugin/common/constants';
+
+import { FtrProviderContext } from '../common/ftr_provider_context';
+import { getUrlPrefix, ObjectRemover } from '../common/lib';
+import { createRule } from './helpers/alerting_api_helper';
+import { createDataView, deleteDataView } from './helpers/data_view';
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const objectRemover = new ObjectRemover(supertest);
+  const es = getService('es');
+
+  describe('Threshold rule data view >', () => {
+    const DATA_VIEW_ID = 'data-view-id';
+
+    let ruleId: string;
+
+    const searchRule = () =>
+      es.search<{ references: unknown }>({
+        index: '.kibana*',
+        query: {
+          bool: {
+            filter: [
+              {
+                term: {
+                  _id: `alert:${ruleId}`,
+                },
+              },
+            ],
+          },
+        },
+        fields: ['alert.params', 'references'],
+      });
+
+    before(async () => {
+      await createDataView({
+        supertest,
+        name: 'test-data-view',
+        id: DATA_VIEW_ID,
+        title: 'random-index*',
+      });
+    });
+
+    after(async () => {
+      objectRemover.removeAll();
+      await deleteDataView({
+        supertest,
+        id: DATA_VIEW_ID,
+      });
+    });
+
+    describe('save data view in rule correctly', () => {
+      it('create a threshold rule', async () => {
+        const createdRule = await createRule({
+          supertest,
+          tags: ['observability'],
+          consumer: 'alerts',
+          name: 'Threshold rule',
+          ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+          params: {
+            criteria: [
+              {
+                aggType: Aggregators.CUSTOM,
+                comparator: Comparator.GT,
+                threshold: [7500000],
+                timeSize: 5,
+                timeUnit: 'm',
+                customMetrics: [
+                  { name: 'A', field: 'span.self_time.sum.us', aggType: Aggregators.AVERAGE },
+                ],
+              },
+            ],
+            alertOnNoData: true,
+            alertOnGroupDisappear: true,
+            searchConfiguration: {
+              query: {
+                query: '',
+                language: 'kuery',
+              },
+              index: DATA_VIEW_ID,
+            },
+          },
+          actions: [],
+        });
+        ruleId = createdRule.id;
+        expect(ruleId).not.to.be(undefined);
+      });
+
+      it('should have correct data view reference before and after edit', async () => {
```

Comment:
@ymao1 I added this test based on your suggestion, please let me know if something is missing :)

Reply:
I would also add an expect for `alertHitsV1` `params.searchConfiguration.index` is set to `kibanaSavedObjectMeta.searchSourceJSON.index` as expected. Otherwise this LGTM!

Reply:
Done!